src/core/public/overlays/sidecar/components/resizable_button.test.tsx

```diff
@@ -106,3 +147,110 @@ test('it should emit onResize with min size when drag if new size is below the m
+
+  // Mock window dimensions
+  Object.defineProperty(window, 'innerHeight', {
+    writable: true,
+    configurable: true,
+    value: 1000,
+  });
+  Object.defineProperty(window, 'innerWidth', {
+    writable: true,
+    configurable: true,
+    value: 1200,
+  });
+
+  // Test takeover mode with size exceeding window height
+  const takeoverProps = {
+    ...props,
+    onResize,
+    dockedMode: SIDECAR_DOCKED_MODE.TAKEOVER,
+  };
+  const takeoverComponent = shallow(<ResizableButton {...takeoverProps} />);
+  const takeoverResizer = takeoverComponent.find(`[data-test-subj~="resizableButton"]`).first();
+
+  takeoverResizer.simulate('mousedown', { clientY: 0, pageX: 0, pageY: 0 });
+  let mouseMoveEvent = new MouseEvent('mousemove', {
+    clientY: -2000,
+    pageX: 0,
+    pageY: 0,
+  } as any);
+  windowEvents?.mousemove(mouseMoveEvent); // Exceeds window.innerHeight
+  windowEvents?.mouseup();
+
+  expect(onResize).not.toHaveBeenCalled();
+
+  // Test right mode with size exceeding window width
+  const rightProps = {
+    ...props,
+    onResize,
+    dockedMode: SIDECAR_DOCKED_MODE.RIGHT,
+  };
+  const rightComponent = shallow(<ResizableButton {...rightProps} />);
+  const rightResizer = rightComponent.find(`[data-test-subj~="resizableButton"]`).first();
+
+  rightResizer.simulate('mousedown', { clientX: 0, pageX: 0, pageY: 0 });
+  mouseMoveEvent = new MouseEvent('mousemove', { clientX: -2000, pageX: 0, pageY: 0 } as any);
+  windowEvents?.mousemove(mouseMoveEvent); // Exceeds window.innerWidth
+  windowEvents?.mouseup();
+
+  expect(onResize).not.toHaveBeenCalled();
+});
+
+test('it should handle window resize events correctly for different docked modes', async () => {
+  restoreWindowEvents();
+  const onResize = jest.fn();
+
+  Object.defineProperty(window, 'innerHeight', {
+    writable: true,
+    configurable: true,
+    value: 1000,
+  });
+  Object.defineProperty(window, 'innerWidth', {
+    writable: true,
+    configurable: true,
+    value: 1200,
+  });
+
+  const takeoverProps = {
+    ...props,
+    onResize,
+    dockedMode: SIDECAR_DOCKED_MODE.TAKEOVER,
+    flyoutSize: 800,
+  };
+  let component = mount(<ResizableButton {...takeoverProps} />);
+
+  await act(async () => {
+    Object.defineProperty(window, 'innerHeight', { value: 600 });
+    window.dispatchEvent(new Event('resize'));
+
+    await new Promise((resolve) => setTimeout(resolve, 50)); // wait for debounce
+  });
+
+  expect(onResize).toHaveBeenCalledWith(600);
+
+  onResize.mockClear();
+
+  const rightProps = {
+    ...props,
+    onResize,
+    dockedMode: SIDECAR_DOCKED_MODE.RIGHT,
+    flyoutSize: 1000,
+  };
+  component = mount(<ResizableButton {...rightProps} />);
+
+  await act(async () => {
+    Object.defineProperty(window, 'innerWidth', { value: 800 });
+    window.dispatchEvent(new Event('resize'));
+
+    await new Promise((resolve) => setTimeout(resolve, 50)); // wait for debounce
+  });
+
+  expect(onResize).toHaveBeenCalledWith(800);
```

Comment:
Can we use `await waitFor` wrap this line and remove the debounce wait in line 231 and 250?

Reply:
Done